base/iterators.jl

```diff
@@ -120,17 +120,9 @@ IteratorSize(::Type{Reverse{T}}) where {T} = IteratorSize(T)
 IteratorEltype(::Type{Reverse{T}}) where {T} = IteratorEltype(T)
 last(r::Reverse) = first(r.itr) # the first shall be last
 
-# reverse-order array iterators: assumes more-specialized Reverse for eachindex
-@propagate_inbounds function iterate(A::Reverse{<:AbstractArray}, state=(reverse(eachindex(A.itr)),))
-    y = iterate(state...)
-    y === nothing && return y
-    idx, itrs = y
-    (A.itr[idx], (state[1], itrs))
-end
-
 # Fallback method of `iterate(::Reverse{T})` which assumes the collection has `getindex(::T) and `reverse(eachindex(::T))`
 # don't propagate inbounds for this just in case
```

Comment:
Probably the comment needs to be updated? I wonder if the bounds-checking wasn't being forwarded for a reason.

Reply:
Hmm, the reason is stated as "just in case". IMO if it's OK to forward for  `AbstractArray` it's also OK to forward for `Any`.

Reply:
```suggestion
```

Reply:
The current version of the PR now just gets rid of the `@propagate_inbounds`, which is consistent with the fact that the method for `AbstractArray` doesn't have `@propagate_inbounds`:

https://github.com/JuliaLang/julia/blob/b265dbac3ef8d1f8f7d779044f458a3c5e2b4689/base/abstractarray.jl#L1235-L1239